### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by rodrigopenayo98**
### Project Description
This Ruby on Rails capstone project involves building a mobile web application for budget management. Users can register and log in to access their private data, add transactions linked to categories, and track their expenses by category.

### Added Features
- User registration and login with mandatory full name, email, and password.
- User authentication to access secured pages.
- Categories page displaying category names, icons, and total transaction amounts.
- Ability to add new categories.
- Transactions page listing transactions by category, with total category amounts.
- Option to add new transactions.
- User-friendly forms for creating categories and transactions.
- Use of Postgres as the database and Devise for authentication.
- Unit and integration tests for critical components.
- Deployed the project online.
- Recorded a 5-minute video showcasing project features and code highlights.
- README documentation with project details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,197 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+
+- We would like to thank Microverse community.
+
```
Comment (Omar-Muhamad) [inline README.md:191]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=of%20the%20design-,requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author,-.%20Therefore%2C%20you%20must), Please gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)

File: app/models/transaction.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Transaction < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :transaction_categories
+  has_many :categories, through: :transaction_categories
+
+  validates :name, presence: true, length: { maximum: 255 }
+  validates :amount, presence: true, numericality: { greater_than: 0, only_integer: true }
+end
```
Comment (Omar-Muhamad) [inline app/models/transaction.rb:8]: - [ ]  As per this [note](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=for%20your%20project-,(please%20note%3A%20%22Transaction%22%20is%20a%20name%20already%20used%20by%20ActiveRecord%2C%20so%20using%20it%20as%20a%20name%20for%20your%20model%20and%20table%20will%20result%20in%20an%20error,-).), "Transaction" is a name already used by ActiveRecord, so using it as a name for your model and table will result in an error Please change it (controller, model, and views). you can name it `payment` instead.

File: app/views/splash_screen/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  <div class="content-splash">
+  <h1 class="title-splash">Welcome to Budget App Rod!</h1>
+
+  <div class="links-container-splash">
+    <%= link_to 'Sign Up', new_user_registration_path, class: 'sign-btn' %>
+    <%= link_to 'Log In', new_user_session_path, class: 'log-btn' %>
+  </div>
+  </div>
+</div>
```
Comment (Omar-Muhamad) [inline app/views/splash_screen/index.html.erb:12]: - [ ] Please add the correct placement of the elements of the splash screen page and stick to the layout and colors of the given design.

| Given Design  | Your Design |
| ------------- | ------------- |
| image | image  |

File: app/views/categories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      </div>
+    </a>
+  <% end %>
+</div>
+
+<div class="add-category-container">
+<a class="add-category" href="<%= new_category_path %>">
+  <button class="btn-add-category">Add a New Category</button>
+</a>
+</div>
```
Comment (Omar-Muhamad) [inline app/views/categories/index.html.erb:32]: - [ ] Please add the correct placement of the `Add a New Category` button make it at the bottom of the page and give it a green color like the given design.

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

File: app/views/transactions/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      </div>
+      
+    <% end %>
+    </div>
+
+  <div class="add-transaction-btn-cont">
+  <a class="add-transaction-a" href="<%= new_category_transaction_path(@category) %>">
+    <button class="add-transaction-btn">Add a New Transaction</button>
+  </a>
+  </div>
```
Comment (Omar-Muhamad) [inline app/views/transactions/index.html.erb:31]: - [ ] Please add the correct placement of the `Add a New Transaction` button make it at the bottom of the page and give it a green color like the given design.

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @rodrigopenayo98 

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by brainconnect93 — COMMENTED** [review]
Hi @rodrigopenayo98,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

- [ ] Kindly note that all the changes requested by the previous reviewer is valid and it will be great if you implement them all. the confusion was from the previous reviewer for making use of approval general comment... So i am invalidating it so that you won't waste a chance of requesting review for a mistake you know nothing about.
so please kindly work on all the changes requested by the reviewer.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

_Cheers and Happy coding!👏👏👏_

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

-----

_Invalid Code Review Request does not count into the code reviews limit._

File: README.md (hunk 1 above)
Comment (rodrigopenayo98) [reply README.md:191]: Done! ✅

File: app/models/transaction.rb (hunk 2 above)
Comment (rodrigopenayo98) [reply app/models/transaction.rb:8]: Done! ✅

File: app/views/splash_screen/index.html.erb (hunk 3 above)
Comment (rodrigopenayo98) [reply app/views/splash_screen/index.html.erb:12]: Done! ✅

File: app/views/transactions/index.html.erb (hunk 5 above)
Comment (rodrigopenayo98) [reply app/views/transactions/index.html.erb:31]: Done! ✅

File: app/views/categories/index.html.erb (hunk 4 above)
Comment (rodrigopenayo98) [reply app/views/categories/index.html.erb:32]: Done! ✅

**Review by binhussen — APPROVED** [review]
Hi @rodrigopenayo98 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Sticked given design ✔️
- well Stractured code✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**